SpokestackTests/TFLiteWakewordRecognizerTest.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  TFLiteWakewordRecognizerTest.swift
+//  SpokestackTests
+//
+//  Created by Noel Weichbrodt on 5/22/20.
+//  Copyright © 2020 Spokestack, Inc. All rights reserved.
+//
+
+import Foundation
+import CryptoKit
+@testable import TensorFlowLite
+@testable import Spokestack
+import XCTest
+
+class TFLiteWakewordRecognizerTest: XCTestCase {
+    
+    func hexStringToData(hexString: String) -> Data? {
+        let len = hexString.count / 2
+        var data = Data(capacity: len)
+        for i in 0..<len {
+            let j = hexString.index(hexString.startIndex, offsetBy: i*2)
+            let k = hexString.index(j, offsetBy: 2)
+            let bytes = hexString[j..<k]
+            if var num = UInt8(bytes, radix: 16) {
+                data.append(&num, count: 1)
+            } else {
+                return nil
+            }
+        }
+        return data
+    }
+    
+    override func setUp() {
+        //let filterHexString = filter.map { String(format: "%02hhx", $0) }.joined()
+        //let filterData = hexStringToData(hexString: MockWakewordModels.filterString)
+        //let _ = FileManager.default.createFile(atPath: MockWakewordModels.filterPath, contents: filterData, attributes: .none)
+    }
+    
+    func testInvoke() {
+        // filter
+        let filter = try! Interpreter(modelPath: MockWakewordModels.filterPath)
+        try! filter.allocateTensors()
+        _ = MockWakewordModels.filterInput.withUnsafeBytes { try! filter.copy(Data($0), toInputAt: 0) }
+        _ = try! filter.invoke()
+        let filterOutput = try! filter.output(at: MockWakewordModels.validIndex)
+        XCTAssertEqual(filterOutput.data, MockWakewordModels.filterOutputData)
+        let filterResult = filterOutput.data.toArray(type: Int32.self, count: filterOutput.data.count/4)
+        XCTAssertEqual(filterResult, MockWakewordModels.filterOutput)
+        
+        // encode
+        let encoder = try! Interpreter(modelPath: MockWakewordModels.encodePath)
+        try! encoder.allocateTensors()
+        _ = MockWakewordModels.encodeInput.withUnsafeBytes { try! encoder.copy(Data($0), toInputAt: 0) }
```

Comment:
Since the three models are run as a pipeline, seems like it would make sense to feed the test output of one model into the input of the next.

Reply:
They are run as a pipeline, and tested as a pipeline, in subsequent tests. Here we just want to assert over each model's expected input and output shapes.